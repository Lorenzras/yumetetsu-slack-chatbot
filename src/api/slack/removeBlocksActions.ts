import {Block} from '@slack/types';
import {slackClient} from '../slackClient';
import {getBlocks} from './getBlocks';

export const removeBlocksActions = async (
  channelId: string,
  ts: string,
) => {
  const blocks = await getBlocks(channelId, ts);
  if (!blocks) return;

  const deletedActionsBlocks = blocks.filter(
    (block) => block.type !== 'actions',
  );

  console.log(deletedActionsBlocks);

  return await slackClient.chat.update({
    channel: channelId,
    ts,
    text: '長期追客',
    blocks: deletedActionsBlocks as Block[],
  });
};
