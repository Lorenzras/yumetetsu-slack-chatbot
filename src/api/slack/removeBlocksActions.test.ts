import {removeBlocksActions} from './removeBlocksActions';

describe('removeActionsFromBlocks', () => {
  it('is successfull', async () => {
    expect(await removeBlocksActions('C02R6P88K7E', '1648092644.383499'))
      .toMatchSnapshot();
  });
});
