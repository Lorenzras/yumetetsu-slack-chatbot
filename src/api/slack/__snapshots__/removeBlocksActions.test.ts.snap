// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeActionsFromBlocks is successfull 1`] = `undefined`;
